From 8fc2d61cd9a09488be9de86830a52e091b45b07b Mon Sep 17 00:00:00 2001
From: Hieu Huynh <hieuh@os.amperecomputing.com>
Date: Wed, 25 Sep 2024 11:07:30 +0000
Subject: [PATCH] Set ApplyTime to Immediate

This set the ApplyTime to Immediate in firmware update POST command.
This should be a temp webui fix until we are comfortable with the
upcoming changes to bmcweb.

This also removes the 0004-Add-default-Target-to-MultipartHttpPush.patch
patch file as this was merged to webui repo.

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>
---
 src/store/modules/Operations/FirmwareStore.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/store/modules/Operations/FirmwareStore.js b/src/store/modules/Operations/FirmwareStore.js
index 78d3b91..ec4fe23 100644
--- a/src/store/modules/Operations/FirmwareStore.js
+++ b/src/store/modules/Operations/FirmwareStore.js
@@ -150,6 +150,8 @@ const FirmwareStore = {
         // when bmcweb is updated
         params.Targets = [`${await this.dispatch('global/getBmcPath')}`];
       }
+      // Set AppyTime to Immediate as workaround
+      params['@Redfish.OperationApplyTime'] = 'Immediate';
       formData.append('UpdateParameters', JSON.stringify(params));
       return await api
         .post(state.multipartHttpPushUri, formData, {
-- 
2.25.1

